v-next/hardhat/src/internal/cli/telemetry/sentry/anonymizer.ts

```diff
@@ -0,0 +1,379 @@
+const ANONYMIZED_MNEMONIC = "<mnemonic>";
+const MNEMONIC_PHRASE_LENGTH_THRESHOLD = 7;
+const MINIMUM_AMOUNT_OF_WORDS_TO_ANONYMIZE = 4;
+
+export class Anonymizer {
+  readonly #configPath?: string;
+
+  constructor(configPath?: string) {
+    this.#configPath = configPath;
+  }
+
+  /**
+   * Given a sentry serialized exception
+   * (https://develop.sentry.dev/sdk/event-payloads/exception/), return an
+   * anonymized version of the event.
+   */
+  public async anonymize(event: any): Promise<either.Either<string, Event>> {
+    if (event === null || event === undefined) {
+      return either.left("event is null or undefined");
+    }
+    if (typeof event !== "object") {
+      return either.left("event is not an object");
+    }
+
+    const result: Event = {
+      event_id: event.event_id,
+      platform: event.platform,
+      timestamp: event.timestamp,
+      extra: event.extra,
+    };
+
+    if (event.exception !== undefined && event.exception.values !== undefined) {
+      const anonymizedExceptions = await this.#anonymizeExceptions(
+        event.exception.values,
+      );
+      result.exception = {
+        values: anonymizedExceptions,
+      };
+    }
+
+    return either.right(result);
+  }
+
+  /**
+   * Return the anonymized filename and a boolean indicating if the content of
+   * the file should be anonymized
+   */
+  public async anonymizeFilename(filename: string): Promise<{
+    anonymizedFilename: string;
+    anonymizeContent: boolean;
+  }> {
+    if (filename === this.#configPath) {
+      const packageJsonPath = await this._getFilePackageJsonPath(filename);
+
+      if (packageJsonPath === null) {
+        // if we can't find a package.json, we just return the basename
+        return {
+          anonymizedFilename: path.basename(filename),
+          anonymizeContent: true,
+        };
+      }
+
+      return {
+        anonymizedFilename: path.relative(
+          path.dirname(packageJsonPath),
+          filename,
+        ),
+        anonymizeContent: true,
+      };
+    }
+
+    const parts = filename.split(path.sep);
+    const nodeModulesIndex = parts.indexOf("node_modules");
+
+    if (nodeModulesIndex === -1) {
+      if (filename.startsWith("internal")) {
+        // show internal parts of the stack trace
+        return {
+          anonymizedFilename: filename,
+          anonymizeContent: false,
+        };
+      }
+
+      // if the file isn't inside node_modules and it's a user file, we hide it completely
+      return {
+        anonymizedFilename: ANONYMIZED_FILE,
+        anonymizeContent: true,
+      };
+    }
+
+    return {
+      anonymizedFilename: parts.slice(nodeModulesIndex).join(path.sep),
+      anonymizeContent: false,
+    };
+  }
+
+  public anonymizeErrorMessage(errorMessage: string): string {
+    errorMessage = this.#anonymizeMnemonic(errorMessage);
+
+    // the \\ before path.sep is necessary for this to work on windows
+    const pathRegex = /\S+[\/\\]\S+/g;
```

Comment:
Note: This is a bit different, it accepts `\` or `/` in any platform. Previously it used `path.sep`.